pkg/server/handlers.go

```diff
@@ -30,6 +35,64 @@ func HostsHandler(w http.ResponseWriter, r *http.Request) {
 	RespondWithJson(w, hostList)
 }
 
+type InstallConfig struct {
+	ClusterName string `json:"clusterName"`
+	DNSDomain   string `json:"DNSDomain"`
+	PullSecret  string `json:"pullSecret"`
+}
+
+type DefineClusterCredentials struct {
+	ClusterName string `json:"clusterName"`
+	DNSDomain   string `json:"DNSDomain"`
+	Username    string `json:"username"`
+	Password    string `json:"password"`
+}
+
+func CreateInstallConfig(w http.ResponseWriter, r *http.Request) {
+	var clusterDefinition DefineClusterCredentials
+	_ = json.NewDecoder(r.Body).Decode(&clusterDefinition)
+
+	tokenRequestURL := fmt.Sprintf("https://developers.redhat.com/auth/realms/rhd/protocol/openid-connect/token")
+
+	resp, err := http.PostForm(tokenRequestURL, url.Values{
+		"grant_type": {"password"},
+		"client_id":  {"uhc"},
+		"username":   {clusterDefinition.Username},
+		"password":   {clusterDefinition.Password},
+	})
+
+	if err != nil {
+		log.Print("Error in api call", err)
+	}
+
+	// Is this needed?
+	// defer resp.Body.Close()
+
+	var tokenData map[string]string
+	json.NewDecoder(resp.Body).Decode(&tokenData)
+
+	if err != nil {
+		RespondWithError(w, err, 500)
+	}
+
+	pullSecretRequestURL := fmt.Sprintf("https://api.openshift.com/api/accounts_mgmt/v1/access_token")
+
+	req, err := http.NewRequest("POST", pullSecretRequestURL, bytes.NewReader(nil))
```

Comment:
Moar error handling. This is one of the most hated features of Go: the constant `if err != nil {}`. But it actually forces you to think about your errors.